packages/ember-data/lib/system/model/states.js

```diff
@@ -685,6 +681,7 @@ var RootState = {
       isDirty: false,
 
       setup: function(internalModel) {
+        internalModel.clearRelationships();
```

Comment:
Hm, I would rely on it there, and not here, but I think we can fix that later when I merge the unload refactor